src/Conversion/ONNXToKrnl/Math/LRN.cpp

```diff
@@ -100,23 +99,27 @@ struct ONNXLRNOpLowering : public ConversionPattern {
               rewriter, loc);
 
           Value sumAlloc = create.mem.alloc(scalarMemRefType);
-          createKrnl.store(
-              emitConstantOp(rewriter, loc, elementType, 0), sumAlloc);
+          createKrnl.store(create.math.constant(elementType, 0), sumAlloc);
 
           // Create the sum reduction loop
-          krnl::BuildKrnlLoop sumLoops(rewriter, loc, 1);
-          sumLoops.createDefineOp();
-          sumLoops.pushBounds(lbMaxList, ubMinList);
-          sumLoops.createIterateOp();
+          std::vector<Value> loop;
+          defineLoops(rewriter, loc, loop, 1);
+          krnl::KrnlIterateOperandPack pack(rewriter, loop);
+          pack.pushIndexExprsBound(lbMaxList);
+          pack.pushIndexExprsBound(ubMinList);
+          KrnlIterateOp iterateOp = createKrnl.iterate(pack);
```

Comment:
@whitneywhtsang is there any reason why we didn't use lamda function to create `krnl.iterate`?

Reply:
Do you mean why we don't use 
```
KrnlBuilder::iterate(ValueRange originalLoops, ValueRange optimizedLoops,
    ValueRange lbs, ValueRange ubs,
    function_ref<void(KrnlBuilder &createKrnl, ValueRange indices)>
        bodyBuilderFn);
```
?
If so, it is because I cannot find an interface that takes `KrnlIterateOperandPack`.

Reply:
Yes, I mean that one. In this case, upper and lower bounds are IndexExpr, so we should use `KrnlBuilder::iterateIE`.

> If so, it is because I cannot find an interface that takes KrnlIterateOperandPack.

`KrnlIterateOperandPack` will be created by the KrnlIterateOp's build by using upper and lower bounds passed to `KrnlBuilder::iterateIE`: https://github.com/onnx/onnx-mlir/blob/main/src/Dialect/Krnl/KrnlOps.cpp#L244

Reply:
`lbMaxList` and `ubMinList` are both `SmallVector<IndexExpr>`, not `IndexExpr`, so cannot use either `KrnlBuilder::iterate` or `KrnlBuilder::iterateIE`.

Reply:
Got it. Thanks!